### PR TITLE
[YUNIKORN-2695] remove core dependency pkg/common

### PR DESCRIPTION
**Pull request by 0lai0**
### What is this PR for?
This PR modifies the dependency between yunikorn-k8shim and yunikorn-core. In this PR, the `common.WaitForCondition` method can be called within yunikorn-k8shim, allowing yunikorn-k8shim to function without relying on yunikorn-core. Consequently, this reduces the dependency of YUNIKORN-shim on YUNIKORN-core.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
[[YUNIKORN-2695]](https://issues.apache.org/jira/browse/YUNIKORN-2695)

### How should this be tested?
make test

### CODE REVIEW

**Review by chenyulin0719 — CHANGES_REQUESTED** [review]
Test failed, could you please fix it. Thanks.

**Comment by 0lai0** [comment]
No problem.I will fix it as soon as possible .

File: pkg/client/clients_test.go [hunk 1]
```diff
@@ -58,16 +58,16 @@ func TestRun(t *testing.T) {
 	}()
 
 	clients.Run(stop)
-	err := common.WaitForCondition(10*time.Millisecond, time.Second, func() bool {
+	err := utils.WaitForCondition(func() bool {
 		return test.RunningInformers.Load() == noOfInformers
-	})
+	}, 10*time.Millisecond, 5*time.Second)
```
Comment (chenyulin0719) [inline pkg/client/clients_test.go:63]: Do there have any reason to change the timeout to 5 secs?

**Review by chenyulin0719 — APPROVED** [review]
LGTM, The 'preemption' e2e test fail was caused by the known issue in https://issues.apache.org/jira/browse/YUNIKORN-2699

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request closed without merging**